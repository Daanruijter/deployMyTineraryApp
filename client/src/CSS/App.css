

.App {
  height:100%;
  display:grid; 
grid-template-rows: 6% 94%;

}

 .hamburger-grid {

grid-row-start: 0;
grid-row-start: 1;
  
} 


.main-content {
  grid-row-start: 1;
grid-row-start: 2;
height:100vh;
}

nav {
  background-color:red !important;
}


