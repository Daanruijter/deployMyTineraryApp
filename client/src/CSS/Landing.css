
.landing-page  {
  
    height: 93%;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }
  
  .logo {
    height: 25%;
    width: 100%;
  }
  
  .start-browsing-flexer {
    display:flex;
    justify-content: center;
    align-items: center;
  }
  
  .startBrowsing
  {
    width:25%;
    
  }
  
  .home-flexer {
    height: 15%;
    width: 100%;
    display:flex;
    justify-content: center;
    align-items: center;
  }
  .homeIcon {
    width: 10%;
  }
  