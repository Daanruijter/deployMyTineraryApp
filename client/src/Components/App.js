import React, { Component } from "react";

import "../CSS/App.css";
import Landing from "./Landing";
import Cities from "./Cities";

import CreateAccount from "./CreateAccount";
import Login from "./Login";
import Favourites from "./Favourites";
import HamburgerMenu from "./HamburgerMenu";
import { Route, Switch } from "react-router-dom";
import HamburgerMenuList from "./HamburgerMenuList";
import Itinerary from "./Itinerary";
import { loadUser } from "../store/actions/authActions";
import { connect } from "react-redux";

class App extends Component {
  componentDidUpdate() {
    if (localStorage.getItem("token")) {
      // console.log("token is there");
    }
    if (!localStorage.getItem("token")) {
      // console.log("token is NOT there");
    }
  }

  componentDidMount() {
    this.props.store.dispatch(loadUser());
  }

  state = {
    hamburgerMenuList: false,
  };

  drawerToggleClickHandler = (e) => {
    this.setState((prevState) => {
      return { hamburgerMenuList: !prevState.hamburgerMenuList };
    });
  };

  backdropClickHandler = () => {
    this.setState({ hamburgerMenuList: false });
  };

  routesGenerator() {
    return (
      <Switch>
        <Route exact path="/" render={(props) => <Landing {...props} />} />
        <Route exact path="/Cities" component={Cities} />
        <Route exact path="/Create-account" component={CreateAccount} />
        <Route exact path="/Login" render={(props) => <Login {...props} />} />
        <Route exact path="/Favourites/:cityName" component={Favourites} />
        <Route
          path="/itinerary/:name/:cityName"
          render={(props) => <Itinerary {...props} />}
        />
      </Switch>
    );
  }

  render() {
    return (
      <div className="App">
        <HamburgerMenu
          drawerToggleClickHandler={this.drawerToggleClickHandler}
        ></HamburgerMenu>

        {/* {localStorage.getItem("token") ? ( */}
        <main className="main-content">
          {/* <BrowserRouter> */}
          {this.state.hamburgerMenuList ? (
            <HamburgerMenuList />
          ) : (
            this.routesGenerator()
          )}
          {/* </BrowserRouter> */}
        </main>
        {/* ) : null} */}
      </div>
    );
  }
}
const mapStateToProps = (state) => {
  return {
    state: state,
  };
};
export default connect(mapStateToProps)(App);
