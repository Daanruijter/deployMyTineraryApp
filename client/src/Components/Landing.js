import React from "react";
// import Itinerary from './Itinerary'

import "../CSS/index.css";

import "../CSS/Landing.css";
import MYtineraryLogo from "../Pictures/MYtineraryLogo.png";
import startBrowsing from "../Pictures/startBrowsing.png";

const Landing = props => {
  return (
    <div className="landing-page">
      <img className="logo" src={MYtineraryLogo} alt="MYtineraryLogo" />
      <p>
        Find your perfect trip, designed by insiders who know and love their
        cities.{" "}
      </p>

      <h1 className="start-browsing-header">Start browsing</h1>
      <div className="start-browsing-flexer">
        <a href="/Cities">
          <img
            className="startBrowsing"
            src={startBrowsing}
            alt="startBrowsing"
          />
        </a>
      </div>
      <p>Want to build your own MYtinerary?</p>
    </div>
  );
};

export default Landing;

// state = {
//     name:'Piet',
//     age: '223'
// }
