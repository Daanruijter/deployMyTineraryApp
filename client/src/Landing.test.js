import React from 'react';
import ReactDOM from 'react-dom'
import { render } from '@testing-library/react';
import App from './App';
import { shallow } from 'enzyme';
import Landing from './Landing';
import toJson from 'enzyme-to-json';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe("what is the unit under test?", () => {
  it("renders without crashing", () => {  
    //write your test here
  });
  it("Landing component should render without crashing", () => {
    //write your test here
  });
  it("Landing component snapshot with Enzyme", () => {  
    //write your test here
  });
});

it('Landing component should render without crashing', () => {
    shallow(<Landing />);
});

it("landing component snapshot with Enzyme", () => {
    const wrapper = shallow(<Landing />);
    expect(toJson(wrapper)).toMatchSnapshot();
});


