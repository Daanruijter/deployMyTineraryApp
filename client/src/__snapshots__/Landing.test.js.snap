// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing component snapshot with Enzyme 1`] = `
<div>
  <img
    alt="MYtineraryLogo"
    className="MYtineraryLogo"
    src="MYtineraryLogo.png"
  />
  <p>
    Find your perfect trip, designed by insiders who know and love their cities. 
  </p>
  <h1
    className="start-browsing-header"
  >
    Start browsing
  </h1>
  <div
    className="start-browsing-flexer"
  >
    <img
      alt="startBrowsing"
      className="startBrowsing"
      src="startBrowsing.png"
    />
  </div>
  <p>
    Want to build your own MYtinerary?
  </p>
  <div
    className="UI-flexer"
  >
    <div
      className="login-box"
    >
      <a
        href=""
      >
        Log in
      </a>
    </div>
    <div
      className="login-box"
    >
      <a
        href=""
      >
        Create Account
      </a>
    </div>
  </div>
  ssss
  <div
    className="home-flexer"
  >
    <img
      alt="homeIcon"
      className="homeIcon"
      src="homeIcon.png"
    />
  </div>
</div>
`;
